src/HexagonOffload.cpp

```diff
@@ -378,10 +277,83 @@ Stmt inject_hexagon_rpc(Stmt s, const Target &host_target) {
         }
     }
 
-    InjectHexagonRpc injector(target);
+    Module hexagon_module("hexagon", target);
+    InjectHexagonRpc injector(hexagon_module);
     s = injector.inject(s);
+
+    if (!hexagon_module.functions().empty()) {
+        *submodules++ = hexagon_module;
```

Comment:
Will this do the right thing if there are multiple submodules being added? The iterator is passed by value so the increment operation won't affect the iterate from the caller, which looks like it assumes that this does increment if necessary.

Maybe just pass a reference to the vector instead?